### PR TITLE
Fix failing main

### PR DESCRIPTION
**Pull request by ephraimbuddy**
When I merged #18724 the jobs ran successfully but it's now failing in main.
This PR fixes it

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

**Review by uranusjr — APPROVED** [review]

**Review by kaxil — APPROVED** [review]

File: airflow/api_connexion/endpoints/task_instance_endpoint.py [hunk 1]
```diff
@@ -326,7 +326,13 @@ def post_set_task_instances_state(*, dag_id: str, session: Session = NEW_SESSION
             detail=f"Task instance not found for task {task_id!r} on execution_date {execution_date}"
         )
 
-    if run_id and not session.query(TI).get({'task_id': task_id, 'dag_id': dag_id, 'run_id': run_id}):
+    if (
+        run_id
+        and session.query(TI)
+        .filter(TI.task_id == task_id, TI.dag_id == dag_id, TI.run_id == run_id)
+        .one_or_none()
+        is None
```
Comment (kaxil) [inline airflow/api_connexion/endpoints/task_instance_endpoint.py:334]: this syntax feels weird, probably round brackets for entire `session.query` would be more readable?

File: airflow/api_connexion/endpoints/task_instance_endpoint.py [hunk 2]
```diff
@@ -326,7 +326,13 @@ def post_set_task_instances_state(*, dag_id: str, session: Session = NEW_SESSION
...
+    ):
```
Comment (kaxil) [inline airflow/api_connexion/endpoints/task_instance_endpoint.py]: nit -- the following is more readable to me but no strong opinion 

```suggestion
    if run_id and (
        session.query(TI)
        .filter(TI.task_id == task_id, TI.dag_id == dag_id, TI.run_id == run_id)
        .one_or_none()
        is None
    ):
```

**Comment by ashb** [comment]
What was the error?

File: airflow/api_connexion/endpoints/task_instance_endpoint.py [hunk 3]
```diff
@@ -326,7 +326,12 @@ def post_set_task_instances_state(*, dag_id: str, session: Session = NEW_SESSION
             detail=f"Task instance not found for task {task_id!r} on execution_date {execution_date}"
         )
 
-    if run_id and not session.query(TI).get({'task_id': task_id, 'dag_id': dag_id, 'run_id': run_id}):
+    if run_id and (
+        session.query(TI)
+        .filter(TI.task_id == task_id, TI.dag_id == dag_id, TI.run_id == run_id)
+        .one_or_none()
+        is None
+    ):
```
Comment (ashb) [inline airflow/api_connexion/endpoints/task_instance_endpoint.py]: tests/api_connexion/endpoints/test_task_instance_endpoint.py::TestPostSetTaskInstanceState::test_should_assert_call_mocked_api_when_run_id: sqlalchemy.exc.InvalidRequestError: Incorrect number of values in identifier to formulate primary key for query.get(); primary key columns are 'task_instance.task_id','task_instance.dag_id','task_instance.run_id','task_instance.map_index'

```suggestion
if run_id and not session.query(TI).get({'task_id': task_id, 'dag_id': dag_id, 'run_id': run_id, 'map_index': -1}):    ):
```

**Review by ashb — CHANGES_REQUESTED** [review]
(Though my suggestion will need formatting correctly)

File: airflow/api_connexion/endpoints/task_instance_endpoint.py (hunk 3 above)
Comment (ephraimbuddy) [reply airflow/api_connexion/endpoints/task_instance_endpoint.py]: Need to pull to fix static checks if any...pulling

**Review by ashb — APPROVED** [review]

**Pull request merged**